source/features/dim-bots.tsx

```diff
@@ -41,7 +41,14 @@ function init(): void {
 
 	// Delay collapsing, but only after they're collapsed on load #5158
 	requestAnimationFrame(() => {
-		select('#repo-content-pjax-container .js-navigation-container')!.classList.add('rgh-dim-bots--after-hover');
```

Comment:
Why not just `?.`?
```js
		select('#repo-content-pjax-container .js-navigation-container')?.classList.add('rgh-dim-bots--after-hover');
Write
```

Reply:
I remember somewhere you said "it just silences error".

Reply:
> I remember somewhere you said "it just silences error".

But that's also what that change does, just in a more verbose way. It should probably be
```ts
if (pageDetect.isMilestoneList()) {
	return;
}

requestAnimationFrame(…)
```

Reply:
> ```ts
> if (pageDetect.isMilestoneList()) {
> 	return;
> }
> ```

This disables the feature on `isMilestoneList` completely. However the error only appears on empty lists.

Reply:
It should be if !container && pageDetect....

Reply:
> However the error only appears on empty lists.

Then how about adding `() => select.exists('.blankslate')` to the feature's `exclude`?

Reply:
> I remember somewhere you said "it just silences error".

[I found that quote](https://github.com//refined-github/refined-github/issues/5300#issuecomment-1013645436), though not by @fregante. Sorry!

Reply:
> [I found that quote](https://github.com//refined-github/refined-github/issues/5300#issuecomment-1013645436), though not by fregante. Sorry!

To be fair, I was only repeating what I learned from similar reviews ;)

> Then how about adding () => select.exists('.blankslate') to the feature's exclude?

Might as well do https://github.com/refined-github/github-url-detection/issues/100. @kidonng do you have time for it? Otherwise I can take it

Reply:
Re: "it just silences the error"

If we *expect* an element to exist, the we must not use `?.` or a conditional check. 

If there situations in which the element can be missing, then we *expect* that, so if/?. are the right solution. 

However, sometimes there are better solutions, like specifically handling or excluding the pages that don't have the item. Blank state is one such page and in #5222 (or the previous one) I suggested a global solution to that.

Reply:
> I suggested a global solution to that

It's here: https://github.com/refined-github/refined-github/issues/4008#issuecomment-822942520

But how can we tell in `runFeature` which errors come from missing elements and which come from broken or outdated selectors?

Reply:
What I suggested there was this logic:

- when an error is thrown, if the page is blank or the repo is archived, ignore the error (or log it as "Page is blank/archived, feature 124 will not work"